### PR TITLE
PE-1137: Fixed wrong behaviour with time picker + release 4.6.5

### PR DESCRIPTION
**Pull request by poblouin**

### CODE REVIEW

File: src/plugins/rangePlugin.ts [hunk 1]
```diff
@@ -178,10 +178,9 @@ function rangePlugin(config: Config = {}): Plugin {
           _prevDates = [...newDates];
         }
 
-        [
-          fp._input.value = "",
-          secondInput.value = "",
-        ] = fp.selectedDates.map(d => fp.formatDate(d, dateFormat));
+        [fp._input.value = "", secondInput.value = ""] = fp.selectedDates.map(
```
Comment (et1421) [inline src/plugins/rangePlugin.ts]: Shouldn't we ignore eslint for those file to more easily find our changes ?

Comment (poblouin) [reply src/plugins/rangePlugin.ts]: We could, but since it's linted now, we will be good in the future :)

File: src/index.ts [hunk 2]
```diff
@@ -2113,9 +2143,9 @@ function FlatpickrInstance(
       const pluginConf = self.config.plugins[i](self) || ({} as Options);
       for (const key in pluginConf) {
         if (HOOKS.indexOf(key as HookKey) > -1) {
-          (self.config as any)[key] = arrayify(
-            pluginConf[key as HookKey] as Hook
-          )
+          (self.config as any)[key] = arrayify(pluginConf[
+            key as HookKey
+          ] as Hook)
```
Comment (poblouin) [inline src/index.ts]: lint

File: src/plugins/rangePlugin.ts [hunk 3]
```diff
@@ -178,10 +178,9 @@ function rangePlugin(config: Config = {}): Plugin {
...
+          d => fp.formatDate(d, dateFormat)
+        );
```
Comment (poblouin) [inline src/plugins/rangePlugin.ts]: lint

File: src/index.ts [hunk 4]
```diff
@@ -1187,10 +1218,9 @@ function FlatpickrInstance(
           : self.config.defaultSeconds
       );
 
-      self.secondElement.setAttribute(
-        "step",
-        self.minuteElement.getAttribute("step") as string
-      );
+      self.secondElement.setAttribute("step", self.minuteElement.getAttribute(
+        "step"
+      ) as string);
```
Comment (poblouin) [inline src/index.ts]: lint

File: src/index.ts [hunk 5]
```diff
@@ -1884,7 +1914,7 @@ function FlatpickrInstance(
    * @param {Boolean} force if true, it will bypass the autoFillDefaultTime
    *                        config and will set the default time anyway
    */
-  function setDefaultTime(force: Boolean = false) {
+  function setDefaultTime(force: boolean = false) {
```
Comment (poblouin) [inline src/index.ts:1973]: lint

File: src/index.ts [hunk 6]
```diff
@@ -202,6 +202,9 @@ function FlatpickrInstance(
 
     setHoursFromInputs();
 
+    // Id the input is empty and the time picker is set to the default time, that means
```
Comment (et1421) [inline src/index.ts]:
```suggestion
    // If the input is empty and the time picker is set to the default time, that means
```

File: src/index.ts [hunk 7]
```diff
@@ -163,6 +163,15 @@ function FlatpickrInstance(
     }
   }
 
+  function isSameTimeAsDefault() {
+    return (
+      self.hourElement &&
+      self.minuteElement &&
+      +self.hourElement.value === self.config.defaultHour &&
```
Comment (et1421) [inline src/index.ts]: Why the + ?

**Review by et1421 — COMMENTED** [review]
The clear button of the time is not working

File: src/index.ts (hunk 7 above)
Comment (poblouin) [reply src/index.ts]: That's the unary operator, this cast the string to a number

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus

Comment (et1421) [reply src/index.ts]: Thank for the education 👌

File: src/index.ts [hunk 8]
```diff
@@ -163,6 +163,25 @@ function FlatpickrInstance(
...
   }
 
+  function getHourElementValueAsMilitary24(): number {
+    const hourElementNumberValue = self.hourElement
+      ? +self.hourElement.value
+      : self.config.defaultHour;
+
+    return self.amPM !== undefined && hourElementNumberValue
+      ? ampm2military(hourElementNumberValue, self.amPM.textContent as string)
+      : hourElementNumberValue;
```
Comment (et1421) [inline src/index.ts:189]: Won't `hourElementNumberValue ` always be true ?

File: src/index.ts [hunk 9]
```diff
@@ -163,6 +163,25 @@ function FlatpickrInstance(
...
+  }
+
+  function isSameTimeAsDefault(): boolean {
+    return (
+      self.hourElement !== undefined &&
+      self.minuteElement !== undefined &&
+      getHourElementValueAsMilitary24() === self.config.defaultHour &&
```
Comment (et1421) [inline src/index.ts]: Does the defaultHour take the am/pm config in consideration ? If not, the condition will be false when it should be true

File: src/index.ts (hunk 8 above)
Comment (poblouin) [reply src/index.ts:189]: It could be equal to 0 thus false and we would return 0

File: src/index.ts (hunk 9 above)
Comment (poblouin) [reply src/index.ts]: No the config should be 24 hours. In ADS we use 0 anyway as per spec so that won't be a problem

File: src/index.ts [hunk 10]
```diff
@@ -163,6 +163,22 @@ function FlatpickrInstance(
...
+  }
+
+  function getMinForType(): Date | undefined {
+    return isTimeInput() ? self.config.minTime : self.config.minDate;
+  }
+
+  function getMaxForType(): Date | undefined {
+    return isTimeInput() ? self.config.maxTime : self.config.maxDate;
+  }
+
```
Comment (et1421) [inline src/index.ts:181]: 👌

**Review by et1421 — APPROVED** [review]

**Pull request merged**